lib/MetaCPAN/Web/Role/Elasticsearch/Adapter.pm

```diff
@@ -0,0 +1,50 @@
+package MetaCPAN::Web::Role::Elasticsearch::Adapter;
+
+use Moose::Role;
+
+=head1 METHODS
+
+=head2 extract_first_element
+
+Elasticsearch 1.x changed the data structure returned when fields are used.
+For example before one could get a ArrayRef[HashRef[Str]] where now
+that will come in the form of ArrayRef[HashRef[ArrayRef[Str]]]
+
+This function will select the first element given an array of hash of array.
+So this:
+
+    [
+      {
+        distribution => ['WhizzBang'],
+        author       => ['EVE'],
+      },
+       ...
+   ]
+
+becomes:
+
+    [
+      {
+        distribution => 'WhizzBang',
+        author       => 'EVE',
+      },
+      ...
+    ]
+
+=cut
+
+sub extract_first_element {
+    my ( $self, $array ) = @_;
+    foreach my $hash ( @{$array} ) {
+        foreach my $field ( %{$hash} ) {
+            my $value = $hash->{$field};
+
+            # We only extract the first element when have an ArrayRef
+            next if not( ref($value) and ( ref($value) eq 'ARRAY' ) );
```

Comment:
I'd really like to avoid 2 branches as much as possible.  If this doesn't get completed in the OPfW term we're looking at another branch which could live as long as the bootstrap branch.  And trying to merge that made me sad.

What I like about this is that we could merge it into master _now_ and not have it sit in a long lived branch.

So, if an adapter is safe for us to use, I'd be more than happy to merge it into master and delete it after the upgrade.  Probably what we really would need to do in the meantime is write a few QA scripts so that we can compare query results on the 0.x and 1.x branches.